README.md

```diff
@@ -1,4 +1,10 @@
-# UCAN Invocation Specification v1.0.0-rc.1
+# UCAN Invocation Specification v1.0.0-rc. 1
+
+FIXME
+- move pipelines, await, etc to own spec
+- explain in FAQ why we don't need `join` (i.e. use promises, since the model is inverted)
+- add expiry
```

Comment:
As "this task is valid for a certain period of time"?
If so, would it be handy to express it at the UCAN level?

I.e., I want this Task invoked and UCAN execute it in that time frame

Reply:
I used to think so too, but we need separate expirations in this version. Because we more rigorously "close the loop", we're not delegating to the executor — the executor is the root issuer. This means that we can no longer infer the expiry from the enclosed delegation.

Reply:
Hmm, I see how delegation to the root issuer doesn't grant him any new capabilities. Yet such delegation may still be useful to express "you can perform that specific attenuated stuff **on my behalf** in **that timeframe**"? An [example usage](https://github.com/ucan-wg/spec/issues/130#issuecomment-1762863709) (`exp` is omitted in those UCANs, but we can imagine that Invoker attenuated to a narrower time range). And so a valid Receipt would need to include explicit delegation from Invoker. :thinking:

Reply:
Happy to learn more about why delegation to the root issuer is not a good approach, btw!

It may not be a good solution for specifying validity timeframe of invocation, if delegation UCAN with that validity timeframe is decoupled from the Invocation / comes out-of-band, as in the example above. Because then even when UCAN expires the Invocation hangs around. User then may issue the same Invocation, and its UCAN can be accedintally used for the previous Invocation as well.

```
at t1, UCAN1 for [t1; t3) -can-> Task <-run- Invocation1
at t3, UCAN2 for [t3; t6) -can-> Task <-run- Invocation2
```
say
at t3, Invocation still wasn't executed, and so it's meant to be considered expired, however
at t3, Issuer issued another Invocation for the same Task
and so
throughout [t3; t6) Executor now can use UCAN2 as authorization to execute both Invocation2 and _Invocation1_!
...Ah, they're actually the same, content-wise, Invocation1 = Invocation2.

Buuut if they would differ, say contained "decision time" - when they've been issued, akin to `iat` that's on Receipt - then Tasks would be the same, yet Invocations would be unique.
```
at t1, UCAN1 for [t1; t3) -can-> Task <-run- Invocation1 iat t1
at t3, UCAN2 for [t3; t6) -can-> Task <-run- Invocation2 iat t3
```

And so Executor could execute both Invocations, whereas Invocation1 is meant to be expired.

Perhaps an interesting solution to that is to have validity timeframe inside the Task.
And so both UCAN and Invocation would be talking about the same stuff.
```
at t1, UCAN1 -can-> Task for [t1; t3) <-run- Invocation1 iat t1
at t3, UCAN2 -can-> Task for [t3; t6) <-run- Invocation2 iat t3
```
and so
at t3, both UCAN1 and Invocation1 are expired
at t5, only UCAN2 and Invocation2 are valid

I.e., move valid timeframe from U of UCAN to CAN (Capability) it's about, so it can be used across both UCAN and Invocation, CAN()
I.e., validity timeframe as attenuation of Capability